### DISCUSSION TITLE
Difficulty with console output

### DISCUSSION THREAD

**Discussion by rgov** [ipxe/ipxe#843] [Q&A] [answered]
I am using iPXE to boot an embedded computer into Alpine Linux over the network. I configured Alpine's kernel command line to provide a serial console on /dev/ttyS0.

This worked fine until I enabled console redirection in my BIOS so that I could use the same serial connection to access the BIOS configuration such as boot order. 

Since I made this change, my iPXE menu *does* still appear over serial, but when I tell it to boot my system, I see the following, and then no further console output either from iPXE or from Linux, though I can tell that Linux is booting up because I can observe network traffic from it.

```
http://10.89.7.28/alpine/vmlinuz-lts... ok
http://10.89.7.28/alpine/initramfs-lts... ok
```

Reading over the [Console documentation](https://ipxe.org/console) I see that I shouldn't set `CONSOLE_SERIAL` if I am using BIOS redirection, so I changed my configuration to

```c
#undef CONSOLE_PCBIOS
#define CONSOLE_PCBIOS CONSOLE_USAGE_ALL

#undef LOG_LEVEL
#define LOG_LEVEL LOG_ALL
```

This didn't cause any apparent change in behavior. I also tried switching off the BIOS console in favor of the serial console, with identical results.

```c
#undef CONSOLE_PCBIOS
#define CONSOLE_SERIAL CONSOLE_USAGE_ALL
```

I was able to get an early boot console from Alpine, [uploaded here](https://gist.github.com/rgov/4fa04e7e43544d3b5ead3c0d9d05d205). I've played around with which `ttyS#` device I set as the console without luck (previously I used `ttyS0`, sometimes Ubuntu likes to call it `ttyS4`, I don't know why it's inconsistent).

**Comment by stappersg** [comment, marked as answer]
And what is expected from the _iPXE community_?

**Comment by NiKiZe** [comment]
iPXE has Serial console support, but if you enable this at the same time as BIOS redirection then you are in for a world of pain, this goes for Linux as well. However, remember that any graphical console would likely kill BIOS redirection.

You would however not want to disable PCBIOS console.
